File: src/WeddingBidders.WebApp/projects/wedding-bidders/src/app/core/models/customer.ts
export interface Customer {
  customerId: string;
  firstname: string;
  lastname: string;
  email: string;
  profileId?: string;
}

export interface RegisterCustomerRequest {
  firstname: string;
  lastname: string;
  email: string;
  password: string;
  confirmPassword: string;
}

export interface RegisterCustomerResponse {
  success: boolean;
  message: string;
  customerId?: string;
}

File: services/wedding/WeddingBidders.Wedding.Core/Model/Customer.cs
using WeddingBidders.Shared.Core;

namespace WeddingBidders.Wedding.Core.Model;

public class Customer : BaseEntity
{
    public Guid CustomerId { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public Guid ProfileId { get; set; }
    public ICollection<Wedding> Weddings { get; set; } = new List<Wedding>();
}
